### ISSUE TITLE
Able to set bounds as other parameters

### ISSUE THREAD

**Issue by mishra-n** [astropy/astropy#16283]
### What is the problem this feature will solve?

I am using Astropy models in MCMC and this would be a useful feature. For example, if I am running an MCMC with my model being two gaussians. The posterior often looks bimodal, when in reality

g1(mu1, sigma) + g2(mu2, sigma) = g1(mu2, sigma) + g2(mu1, sigma)

and the sampler is just checking two equivalent models. A workaround would be to set mu2 to always be greater than mu1 which breaks the degeneracy, but I don't think there is a way to do this in astropy models

### Describe the desired outcome

A way to set the bounds of a variable in astropy model to be another parameter. This way, the bounds of mu1 will be (-inf, mu2), this way the order of the two gaussians is preserved.

**Comment by pllim** [comment]
@adrn , I vaguely remember you had some experience in such things. Did I misremember or can you advise here? Thanks!

**Comment by adrn** [comment]
Yes, but only in the sense that I have strong opinions from my own uses 😅 ! `astropy.modeling` is not a fully-fledged probabilistic programming language and I don't think it will (or should) grow to be one. If you need finer scale control over parameters and their prior distributions (which is what you are asking about here), I would strongly recommend using a more flexible, JIT-able framework like [numpyro](https://github.com/pyro-ppl/numpyro) or [pymc](https://www.pymc.io/welcome.html).
